File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int maxDepth(struct TreeNode* root) {
    if (root == NULL){
        return 0;
    }
    int left_len = maxDepth(root->left);
    int right_len = maxDepth(root->right);
    int max;
    if (left_len > right_len){
        max = left_len;
    } else { max = right_len; }
    return max + 1;
    
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        node = root
        if node is None:
            return 0
        left_length = self.maxDepth(node.left)
        right_length = self.maxDepth(node.right)
        return max(left_length, right_length) + 1
